packages/react-aria-components/src/Select.tsx

```diff
@@ -141,45 +122,56 @@ function Select<T extends object>(props: SelectProps<T>, ref: ForwardedRef<HTMLD
   delete DOMProps.id;
 
   return (
-    <Provider
-      values={[
-        [InternalSelectContext, {state, valueProps, placeholder: props.placeholder}],
-        [LabelContext, {...labelProps, ref: labelRef, elementType: 'span'}],
-        [ButtonContext, {...triggerProps, ref: buttonRef, isPressed: state.isOpen}],
-        [PopoverContext, {
-          state,
-          triggerRef: buttonRef,
-          preserveChildren: true,
-          placement: 'bottom start',
-          style: {'--trigger-width': buttonWidth} as React.CSSProperties
-        }],
-        [ListBoxContext, {state, [slotCallbackSymbol]: setListBoxProps, ...menuProps}],
-        [TextContext, {
-          slots: {
-            description: descriptionProps,
-            errorMessage: errorMessageProps
-          }
-        }]
-      ]}>
-      <div
-        {...DOMProps}
-        {...renderProps}
-        {...focusProps}
-        ref={ref}
-        slot={props.slot}
-        data-focused={state.isFocused || undefined}
-        data-focus-visible={isFocusVisible || undefined}
-        data-open={state.isOpen || undefined}
-        data-disabled={props.isDisabled || undefined}
-        data-validation-state={props.validationState || undefined}
-        data-required={props.isRequired || undefined} />
-      {portal}
-      <HiddenSelect
-        state={state}
-        triggerRef={buttonRef}
-        label={label}
-        name={props.name} />
-    </Provider>
+    <>
+      {/* Render a hidden copy of the children so that we can build the collection even when the popover is not open.
+        * This should always come before the real DOM content so we have built the collection by the time it renders during SSR. */}
+      <Hidden>
+        <Provider
+          values={[
+            [InternalSelectContext, {state, valueProps, placeholder: props.placeholder}],
+            [ListBoxContext, {document}]
```

Comment:
Don't we have to provide `items` the ListBox here as well like we do for ComboBox?

Reply:
hmm interesting. I wonder if we should even support `items` as a prop on the top-level `Select` or if `items` on the `ListBox` is enough? For ComboBox we needed to support `defaultItems` which isn't a prop of `ListBox`. Should `Select` match ComboBox or do we treat that as a special case? Sorta weird to have `items` in two places.

Reply:
my gut feeling here is that we should get rid of `items` on the top level `Select` if possible, it feels more like a carry over from RSP's api that isn't needed here since the user provides the ListBox in RAC and I agree that having `items` in two places is pretty confusing.